File: export_bbs_fcn.py
import os
from shutil import copyfile

def export_bbs_fcn(SA, param):
    from PyQt6.QtWidgets import QFileDialog

    # Request the files to load
    dialog = QFileDialog()
    dialog.setFileMode(QFileDialog.FileMode.ExistingFiles)
    dialog.setNameFilter("BBS Files (*.bbs)")
    dialog.setDirectory(os.path.dirname(SA['fullfilename']))

    if dialog.exec():
        files = dialog.selectedFiles()
    else:
        print("Abort")
        return

    for file in files:
        with open(file, 'r') as f:
            BBStxt = f.readlines()

        # Search the content for selected lines
        TF = [0]
        for si, selected in enumerate(SA['Table']['Selected']):
            if selected and any(SA['Table']['Filename'][si] in line for line in BBStxt):
                TF.append(next(i for i, line in enumerate(BBStxt) if SA['Table']['Filename'][si] in line))

        # Backup the original file
        backup_file = f"{file}_orig"
        copyfile(file, backup_file)

        # Write the new file
        with open(file, 'w') as f:
            for index in TF:
                f.write(BBStxt[index])

File: Export_BBS_Fcn.m
function Export_BBS_Fcn(eventdata, data)

global SA param player


% Read original BBS file
% Find corrosponding line
% write new file


% Read Original file

% request the files to load
[filename, pathname] = uigetfile('*.bbs', ...
   'Pick file to be extracted', fileparts(SA.fullfilename), 'MultiSelect', 'on')

% Check inputs
if iscell(filename)
   nfiles = size(filename,2);
   files = filename;
elseif isnumeric(filename)
   disp('abort')
   return
else
   nfiles = 1;
   files{1} = filename;
end

for fi = 1:length(files)
    BBStxt = readlines(fullfile(pathname, filename{fi}));
    
    % Now search the content for selected lines
    
    TF=[];
    TF(1) = 1;
    i=1;
    for si = 1:length(SA.Table.Selected)
        if SA.Table.Selected{si} & ~isempty(find(contains(BBStxt, SA.Table.Filename{si})))
            i = i+1;
            TF(i) = find(contains(BBStxt, SA.Table.Filename{si}));
        end
    end
    
    copyfile(fullfile(pathname, filename{fi}), [fullfile(pathname, filename{fi}),'_orig'])
    
    fid = fopen(fullfile(pathname, filename{fi}),'w');
    fprintf(fid, "%s\n", BBStxt{TF})
    fclose(fid)
    
end